### PR TITLE
perceptron predictor

### PR DESCRIPTION
**Pull request by sam-ginsburg**

### CODE REVIEW

File: bpred.c [hunk 1]
```diff
@@ -104,6 +104,12 @@ bpred_create(enum bpred_class class,    /* type of predictor to create */
 
     break;
 
+  case BPredPercept:
```
Comment (stevekuznetsov) [inline bpred.c]: I agree that the last three letters of perceptron were just too heinous to add to the name :laughing:

File: bpred.c [hunk 2]
```diff
@@ -683,13 +753,17 @@ bpred_dir_lookup(struct bpred_dir_t *pred_dir,  /* branch dir predictor inst */
...
+      return (char *)p2;
+      break;
     case BPred6bit:
     case BPred5bit:
     case BPred4bit:
     case BPred3bit:
     case BPred2bit:
     case BPred1bit:
-      p = &pred_dir->config.bimod.table[BIMOD_HASH(pred_dir, baddr)];
+      p = pred_dir->config.bimod.table[BIMOD_HASH(pred_dir, baddr)];
```
Comment (stevekuznetsov) [inline bpred.c]: Why this change?

Comment (sam-ginsburg) [inline bpred.c]: This was a mistake that shouldn't have been included. I'll fix it.

**Comment by stevekuznetsov** [comment]
@sam-ginsburg you're going to need to rebase on master and fix merge conflicts - probably in the benchmark outputs:

``` sh
$ git checkout master
$ git fetch origin
$ git merge origin/master
$ git push sam-ginsburg master
$ git checkout add-perceptron
$ git rebase master
```

**Comment by stevekuznetsov** [comment]
@sam-ginsburg Once you run

``` sh
$ git rebase master
```

There will be conflicts. Fix them as usual then run

``` sh
$ git rebase --continue
$ git push sam-ginsburg add-perceptron
```
